externs/shaka/abr_manager.js

```diff
@@ -92,9 +92,11 @@ shaka.extern.AbrManager = class {
    * @param {number} numBytes The total number of bytes transferred.
    * @param {boolean} allowSwitch Indicate if the segment is allowed to switch
    *     to another stream.
+   * @param {shaka.extern.Request=} request
+   *     A reference to the request
    * @exportDoc
    */
-  segmentDownloaded(deltaTimeMs, numBytes, allowSwitch) {}
+  segmentDownloaded(deltaTimeMs, numBytes, allowSwitch, request) {}
```

Comment:
I don't see this new parameter added to our default AbrManager or any test fakes.

Reply:
We are using a custom ABR manager and are experimenting with different algorithms and I didn't want to complicate or change the behaviour simple ABR manager.

Reply:
Can you explain the algorithm? Maybe can be interesting in Shaka…

Reply:
Its basically to mitigate the reasons outlined in this comment https://github.com/shaka-project/shaka-player/blob/d2516498d2add42cf3157b9a6b0a3fa475ade653/lib/abr/ewma_bandwidth_estimator.js#L54

Reply:
> > I don't see this new parameter added to our default AbrManager or any test fakes.
> I didn't want to complicate or change the behaviour of the simple ABR manager.

You don't need to implement the behavior.  Please just add the argument, so that the method signature of our builtins always agree with the external definition.

Reply:
Done!